jira/client.py

```diff
@@ -888,6 +893,46 @@ def create_issue(self, fields=None, prefetch=True, **fieldargs):
         else:
             return Issue(self._options, self._session, raw=raw_issue_json)
 
+    def create_issues(self, field_list, prefetch=True):
+        """Bulk create new issues and return an issue Resource for each successfully created issue.
+
+        See `create_issue` documentation for field information.
+
+        :param field_list: a list of dicts each containing field names and the values to use. Each dict
+            is an individual issue to create and is subject to its minimum requirements.
+        """
+        data = {'issueUpdates': []}
+        for field_dict in field_list:
+            issue_data = _field_worker(field_dict)
+            p = issue_data['fields']['project']
+
+            if isinstance(p, string_types) or isinstance(p, integer_types):
+                issue_data['fields']['project'] = {'id': self.project(p).id}
+
+            p = issue_data['fields']['issuetype']
+            if isinstance(p, integer_types):
+                issue_data['fields']['issuetype'] = {'id': p}
+            if isinstance(p, string_types) or isinstance(p, integer_types):
+                issue_data['fields']['issuetype'] = {'id': self.issue_type_by_name(p).id}
+
+            data['issueUpdates'].append(issue_data)
+
+        url = self._get_url('issue/bulk')
+        r = self._session.post(url, data=json.dumps(data))
+
+        raw_issue_json = json_loads(r)
+        issue_list = []
+        for error in raw_issue_json['errors']:
+            failed_fields = field_list[error['failedElementNumber']]
+            logging.error('Error creating issue with fields: %s\n%s', failed_fields,
+                          error['elementErrors']['errors'])
+        for result in raw_issue_json['issues']:
+            if prefetch:
+                issue_list.append(self.issue(result['key']))
+            else:
+                issue_list.append(Issue(self._options, self._session, raw=result))
+        return issue_list
```

Comment:
This design is problematic, because the caller has no easy way to correlate the inputs to the outputs.

i.e. if I input a list of 3 issues and get back 2 results, how do I know which two inputs correspond to those two results?